### DISCUSSION TITLE
Why is the ORM in Goravel not very similar to Eloquent in Laravel

### DISCUSSION THREAD

**Discussion by acetex** [goravel/goravel#439] [General]
Why is the ORM in Goravel not very similar to Eloquent in Laravel but instead resembles Sequelize, TypeORM in Node.js, or the ORM in .NET Core? Shouldn't the usage of a good ORM be like this?

Example differences:

Laravel:
$user = User::where('name', 'like', '%John%')->get();
$user[2]->name = 'John Doe';
$user[2]->save();

$user[2]->delete();

I don't know how to write code like this in Goravel. Does Goravel still maintain the Active Record pattern like Eloquent? If there is a way to do this in a similar manner, why wasn't it designed to be more like Eloquent? For example, Lucid ORM in TypeScript is very similar to Eloquent, unlike TypeORM or Sequelize.

**Comment by hwbrzzl** [comment]
It's decided by the Golang language features, we can not totally implement Eloquent like Laravel, but we are looking forward to that, and glad to see a PR to optimize it.
